localstack/services/apigateway/templates.py

```diff
@@ -87,16 +87,15 @@ def urlDecode(self, s):
         return unquote_plus(s)
 
     def escapeJavaScript(self, s):
-        try:
-            return json.dumps(json.loads(s))
-        except Exception:
-            primitive_types = (str, int, bool, float, type(None))
-            s = s if isinstance(s, primitive_types) else str(s)
-        if str(s).strip() in {"true", "false"}:
-            s = bool(s)
-        elif s not in [True, False] and is_number(s):
-            s = to_number(s)
-        return json.dumps(s)
+        """
+        Escapes a string that will turn any regular single quotes (') into escaped ones (\').
+        JSON dumps will escape the single quotes.
+        https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-mapping-template-reference.html
+        """
+        # empty string escapes to empty object
+        if isinstance(s, str) and len(s.strip()) == 0:
+            return "{}"
+        return s if is_number(s) else json.dumps(s)[1:-1]
```

Comment:
I feel like we should consider a few more corner cases here, right? What if `s` is `None`, then the result returned would be the string `"ul"` (trimmed from `"null"`)

Similar for the case where `s` is `True` or `False` - it just happens to be the case that `is_number(...)` also treats booleans as numbers (not really accurate by the way), but would be better to make these cases explicit in here...?